core/views.py

```diff
@@ -1 +1,39 @@
-# Create your views here.
+from django.shortcuts import get_object_or_404
+from django.views.generic import ListView, TemplateView
+
+from .models import Post
+
+
+class PostListlView(ListView):
+
+    """Provide a temporary view for internal testing of the post view count feature.
+
+    This view is not intended for production use and should be accessed only by developers.
+    """
+
+    model = Post
+    template_name = "post-list.html"
+    context_object_name = "posts"
+
+
+class PostDetailView(TemplateView):
+
+    """Provide a temporary view for internal testing of the post view count feature.
+
+    This view is not intended for production use and should be accessed only by developers.
+    """
+
+    template_name = "post-detail.html"
+
+    def get_context_data(self: "PostDetailView", **kwargs: dict) -> dict:
+        context = super().get_context_data(**kwargs)
+
+        post_id = self.kwargs.get("pk")
+        post = get_object_or_404(Post, pk=post_id)
+        post.view_count += 1
+        post.save(update_fields=["view_count"], skip_version_check=True)
```

Comment:
This should be a method on post itself, not in the view. The method itself is ok, but really, should be in model - after all it's logic on the model. 
(lines 31 to 34)

Then, the call to post.update_display_counter() should be most likely called in view get_object method, so tou will not even need get_context_data.